src/test/java/com/thealgorithms/sorts/SortUtilsTest.java

```diff
@@ -67,4 +73,23 @@ void isSortedListFalse() {
         List<Integer> array3 = List.of(5, 4, 3, 2, 1);
         assertFalse(SortUtils.isSorted(array3));
     }
+
+    @ParameterizedTest
+    @MethodSource("provideArraysForSwap")
+    public <T> void testSwap(T[] array, int i, int j, T[] expected, String message) {
+        SortUtils.swap(array, i, j);
+        assertArrayEquals(expected, array, message);
+    }
+
+    private static Stream<Arguments> provideArraysForSwap() {
+        return Stream.of(Arguments.of(new Integer[] {1, 2, 3, 4}, 1, 2, new Integer[] {1, 3, 2, 4}, "Swapping adjacent elements should work correctly."), Arguments.of(new Integer[] {1, 2, 3, 4}, 0, 3, new Integer[] {4, 2, 3, 1}, "Swapping non-adjacent elements should work correctly."),
+            Arguments.of(new Integer[] {1, 2, 3, 4}, 2, 2, new Integer[] {1, 2, 3, 4}, "Swapping the same index should not change the array."), Arguments.of(new String[] {"a", "b", "c", "d"}, 0, 3, new String[] {"d", "b", "c", "a"}, "Swapping first and last elements should work correctly."),
+            Arguments.of(new String[] {null, "b", "c", null}, 0, 3, new String[] {null, "b", "c", null}, "Swapping null elements should work correctly."), Arguments.of(new Integer[] {}, 0, 0, new Integer[] {}, "Swapping in an empty array should not throw an error."));
+    }
```

Comment:
I would be careful with the messages:
- _Swapping the same index should not change the array._ - this test does not check that (i.e. the _swap_ could still happen, but the test would pass),
-  _Swapping in an empty array should not throw an error._ - this is an accident.

So from my side: feel free to remove them.

Reply:
Good point. They are removed. From such perspective they are ambiguous.